src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java

```diff
@@ -248,75 +250,78 @@ private CompletableFuture<Boolean> checkExistingOrder(CompositePurchaseOrder com
-    logger.debug("getExistingOrderById:: Trying to retrieve existing Order with ID: {}", compositePurchaseOrder.getId());
+    LOGGER.debug("getExistingOrderById:: Trying to retrieve existing Order with ID: {}", compositePurchaseOrder.getId());
     String endpoint = String.format(ORDERS_BY_ID_ENDPOINT, compositePurchaseOrder.getId());
     return restClient.handleGetRequest(endpoint).toCompletionStage().toCompletableFuture()
       .thenCompose(order -> {
         CompositePurchaseOrder compPO = order.mapTo(CompositePurchaseOrder.class);
         String poLineNumber = compPO.getCompositePoLines().get(FIRST_ELEM).getPoLineNumber();
         if (StringUtils.isEmpty(poLineNumber)) {
-          logger.warn("getExistingOrderById:: No PO Line Number found for existing Order with ID: {}", compositePurchaseOrder.getId());
+          LOGGER.warn("getExistingOrderById:: No PO Line Number found for existing Order with ID: {}", compositePurchaseOrder.getId());
           return completedFuture(compositePurchaseOrder);
         }
         compositePurchaseOrder.getCompositePoLines().get(FIRST_ELEM).setPoLineNumber(poLineNumber);
         return completedFuture(compositePurchaseOrder);
       })
       .exceptionally(t -> {
-        logger.error("Error looking up for existing PO Line Number: {}", compositePurchaseOrder.getPoNumber(), t);
+        LOGGER.error("Error looking up for existing PO Line Number: {}", compositePurchaseOrder.getPoNumber(), t);
         return compositePurchaseOrder;
       });
 
   }
+
   public Void handleError(Throwable throwable) {
-    final javax.ws.rs.core.Response result;
+    Throwable cause = throwable.getCause();
+    LOGGER.error("Exception placing order", cause);
+    asyncResultHandler.handle(Future.succeededFuture(mapExceptionToResponse(cause != null ? cause : throwable)));
+    return null;
+  }
 
-    logger.error("Exception placing order", throwable.getCause());
-    GobiResponse response = new GobiResponse();
-    response.setError(new ResponseError());
+  private static Response mapExceptionToResponse(Throwable t) {
+
+    String message = t.getMessage();
 
-    final Throwable t = throwable.getCause();
     if (t instanceof HttpException) {
-      final int code = ((HttpException) t).getCode();
-      final String message = t.getMessage();
-
-      switch (code) {
-      case 400:
-        response.getError().setCode(CODE_BAD_REQUEST);
-        response.getError().setMessage(HelperUtils.truncate(t.getMessage(), 500));
-        result = respond400WithApplicationXml(GobiResponseWriter.getWriter().write(response));
-        break;
-      case 500:
-        result = respond500WithTextPlain(message);
-        break;
-      case 401:
-        result = respond401WithTextPlain(message);
-        break;
-      default:
-        result = respond500WithTextPlain(message);
+      switch (((HttpException) t).getCode()) {
```

Comment:
these style can be improved by using java 17 style

Reply:
check all import settings, especially their order

Reply:
Thank you. The only difference now is that currentl java.* is followed by javax.* but in your screenshots they have a reverse order. Does it make sense to reformat those?

Reply:
javax.
java.
is correct one.

Reply:
I am not fan of these style to be honest. it is decided by folio based on Eclipse idea a while ago. I use this import style to be more compatible with all modules